app/components/chart/header.js

```diff
@@ -114,6 +122,33 @@ var TidelineHeader = React.createClass({
       </div>
       );
     /* jshint ignore:end */
+  },
+
+  /**
+   * Helper function for rendering the various navigation buttons in the header.
+   * It accounts for the transition state and disables the button if it is currently processing.
+   * 
+   * @param  {String} buttonClass
+   * @param  {Function} clickAction
+   * @param  {String} icon
+   * 
+   * @return {ReactElement}
+   */
+  renderNavButton: function(buttonClass, clickAction, icon) {
+    var nullAction = function(e) {
+      if (e) {
+        e.preventDefault();
+      }
+    };
+    if (this.props.inTransition) {
+      /* jshint ignore:start */
```

Comment:
Yea. Perhaps it makes sense to merge that branch last, after merging this and all others we can then remove all ignore flags. Thoughts?